doc/arch/014-reporting-etl-pipeline.md

```diff
@@ -0,0 +1,90 @@
+updating patient info, or tracking overdue patients. As such, reporting queries grow complex. Asking questions like
+"latest BP in a month", "prescribed drugs as of a date", "current step of treatment protocol" involve complex clauses
+in the SQL queries.
+
+This complexity has a tangible impact on the quality of our reports. We risk data inconsistency as complex queries are
+more error-prone when reused in multiple places.
+
+
+### 2. Coupling
+
+When the application schema is used for reporting also, there is a tight coupling between application requirements and
+reporting requirements. Changes to the schema to optimize for reporting may have a negative impact on complexity in the
+application, and vice versa.
+
+In addition to complexity, performance is also a growing concern as we scale. Heavy reporting operations may strain the
+application's ability to write data to the database. Conversely, heavy application activity may impact our ability to
+generate reports in a performant manner. We currently use heavy layers of caching to mitigate these performance
+concerns, but the less caching we _need_ to do, the better.
+
+### Priorities and constraints
+
+The upcoming changes to our reporting pipeline are quite new to us, so the priority is the ability to iterate.
+Performance is a secondary concern, since we are not yet at a scale where we need to worry about it too much, and our
+caching strategy works well today.
+
+## Decision
+
+We will introduce a set or reporting tables, implemented as database views, to start powering our reporting
+requirements. This reporting schema will transform the data in our application tables into a more reports-friendly
+format, exposing convenient building blocks to build reports with straightforward SQL queries.
+
+Once the reporting tables are populated, we will use Metabase to create and maintain the SQL queries that run against
+the reporting tables to generate reports.
+
+```
+ --------------------             ------------------------------------             ------------------
+| Application tables |    -->    | Reporting tables as database views |    -->    | Metabase reports |
+ --------------------             ------------------------------------             ------------------
+eg. blood_pressures              eg. blood_pressures_over_time                    eg. monthly control rate query
+                                 (latest BP for each patient as of every month)   (as of every month, how many BPs were
+                                                                                   controlled and recent)
+```
+
+The reporting tables will be implemented as database views within the Simple database.
+
+We will use Metabase for the reports.
+* Easy-to-use interface
+* Easy to grant access to stakeholders, and controlled access to ourselves if necessary
+* Dynamic queries (eg. pick your own date range)
+
```

Comment:
Looking at #2330, it seems like `calendar_month` will be a centrepiece for how we handle time. Should we include some details about what our time abstraction might look like?